packages/good-design/src/advanced/web3action/Web3Action.tsx

```diff
@@ -108,9 +108,13 @@ export const Web3ActionButton: FC<Web3ActionProps> = withTheme({
       if (!supportedChains.includes(chainId ?? 0)) {
         setActionText(ButtonSteps.switch);
         await switchToChain(supportedChains[0]);
+        setActionText(ButtonSteps.action);
         return;
       }
 
+      // if there is already a switch request, don't do action
+      if (actionText === ButtonSteps.switch) return;
+
```

Comment:
this code is needed only for this specific case? or is this a general fix?

Reply:
I'd suggest to set exact flag regarding "if there is already a switch request" state. for example use some ref

and of course, to check is already switching **before** switchToChain call